src/Common/src/Microsoft/Win32/SafeHandles/SafeHeapHandle.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Win32.SafeHandles
+{
+    using System;
+    using System.Diagnostics;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// Handle for heap memory
+    /// </summary>
+    /// <remarks>
+    /// Uses new Heap* methods instead of Local* methods, which are deprecated.  While both calls utilize the same underlying
+    /// heap allocation, Local* adds some overhead (*significant* overhead if LMEM_MOVEABLE is used). .NET forces LMEM_FIXED
+    /// with LocalAlloc for Marshal.AllocHGlobal so it doesn't hit the super slow path.
+    /// 
+    /// Windows attempts to grab space from the low fragmentation heap if the requested memory is below a platform specific
+    /// threshold and certain flags aren't in play (such as NO_SERIALIZE).
+    /// </remarks>
+    internal sealed class SafeHeapHandle : SafeBuffer
+    {
+        /// <summary>
+        /// The handle for the process heap.
+        /// </summary>
+        private static readonly IntPtr s_processHeap = Interop.mincore.GetProcessHeap();
```

Comment:
Have you measured the performance benefits of using these HeapAlloc functions vs just using Marshal.AllocHGlobal?  With all of the pooling we're doing, are the benefits really measurable?  If not, I'd prefer to see us just use the existing functions, and then this would work cross-platform without having to do any special per-platform implementation.